core/services/vrf/delegate.go

```diff
@@ -218,72 +375,96 @@ func (lsn *listener) run(unsubscribeLogs func(), minConfs uint32) {
 					lsn.l.ErrorIf(lsn.logBroadcaster.MarkConsumed(lsn.db, lb), "failed to mark consumed")
 					continue
 				}
-
-				s := time.Now()
-				vrfCoordinatorPayload, req, err := lsn.ProcessLog(req, lb)
-				f := time.Now()
-				err = postgres.GormTransactionWithDefaultContext(lsn.db, func(tx *gorm.DB) error {
-					if err == nil {
-						// No errors processing the log, submit a transaction
-						var etx bulletprooftxmanager.EthTx
-						var from common.Address
-						from, err = lsn.gethks.GetRoundRobinAddress()
-						if err != nil {
-							return err
-						}
-						etx, err = lsn.txm.CreateEthTransaction(tx,
-							from,
-							lsn.coordinator.Address(),
-							vrfCoordinatorPayload,
-							lsn.cfg.EthGasLimitDefault(),
-							&models.EthTxMetaV2{
-								JobID:         lsn.job.ID,
-								RequestID:     req.RequestID,
-								RequestTxHash: lb.RawLog().TxHash,
-							},
-							bulletprooftxmanager.SendEveryStrategy{},
-						)
-						if err != nil {
-							return err
-						}
-						// TODO: Once we have eth tasks supported, we can use the pipeline directly
-						// and be able to save errored proof generations. Until then only save
-						// successful runs and log errors.
-						_, err = lsn.pipelineRunner.InsertFinishedRun(tx, pipeline.Run{
-							State:          pipeline.RunStatusCompleted,
-							PipelineSpecID: lsn.job.PipelineSpecID,
-							Errors:         []null.String{{}},
-							Outputs: pipeline.JSONSerializable{
-								Val: []interface{}{fmt.Sprintf("queued tx from %v to %v txdata %v",
-									etx.FromAddress,
-									etx.ToAddress,
-									hex.EncodeToString(etx.EncodedPayload))},
-							},
-							Meta: pipeline.JSONSerializable{
-								Val: map[string]interface{}{"eth_tx_id": etx.ID},
-							},
-							CreatedAt:  s,
-							FinishedAt: null.TimeFrom(f),
-						}, nil, false)
-						if err != nil {
-							return errors.Wrap(err, "VRFListener: failed to insert finished run")
-						}
-					}
-					// Always mark consumed regardless of whether the proof failed or not.
-					err = lsn.logBroadcaster.MarkConsumed(tx, lb)
-					if err != nil {
-						return err
-					}
-					return nil
+				confirmedAt := lsn.getConfirmedAt(req, minConfs)
+				lsn.reqsMu.Lock()
+				lsn.reqs = append(lsn.reqs, request{
+					confirmedAtBlock: confirmedAt,
+					req:              req,
+					lb:               lb,
 				})
-				if err != nil {
-					lsn.l.Errorw("VRFListener failed to save run", "err", err)
-				}
+				lsn.reqAdded()
+				lsn.reqsMu.Unlock()
 			}
 		}
 	}
 }
 
+func (lsn *listener) getConfirmedAt(req *solidity_vrf_coordinator_interface.VRFCoordinatorRandomnessRequest, minConfs uint32) uint64 {
+	newConfs := uint64(minConfs) * (1 << lsn.respCount[req.RequestID])
+	if newConfs > 200 {
```

Comment:
hm i think this was already here, but might be good to add a comment or a named constant for why 200